ui/app/src/components/shared/ResourceCard.tsx

```diff
@@ -72,24 +81,31 @@ export const ResourceCard: React.FunctionComponent<ResourceCardProps> = (props:
     },
   });
 
+  let canConnect = props.resource.properties && props.resource.properties['connection_uri'];
+  let connectUri = canConnect ? props.resource.properties['connection_uri'] : '';
+
   return (
     <>
       <Stack style={cardStyles}>
         <Stack horizontal>
-        <Stack.Item grow={5} style={headerStyles}>
-          <Link to={props.resource.resourcePath} onClick={() => {props.selectResource(props.resource); return false}} style={headerLinkStyles}>{props.resource.properties.display_name}</Link>
-        </Stack.Item>
-        <Stack.Item style={headerIconStyles}>
-          <Stack horizontal>
-            <Stack.Item><IconButton iconProps={{iconName: 'Info'}} id={`item-${props.itemId}`} onClick={() => setShowInfo(!showInfo)} /></Stack.Item>
-            <Stack.Item>{props.contextMenuElement && props.contextMenuElement}</Stack.Item>
-          </Stack>
-          
-        </Stack.Item>
+          <Stack.Item grow={5} style={headerStyles}>
+            <Link to={props.resource.resourcePath} onClick={() => {props.selectResource(props.resource); return false}} style={headerLinkStyles}>{props.resource.properties.display_name}</Link>
+          </Stack.Item>
+          <Stack.Item style={headerIconStyles}>
+            <Stack horizontal>
+              <Stack.Item><IconButton iconProps={{iconName: 'Info'}} id={`item-${props.itemId}`} onClick={() => setShowInfo(!showInfo)} /></Stack.Item>
+              <Stack.Item>{props.contextMenuElement && props.contextMenuElement}</Stack.Item>
+            </Stack>
+          </Stack.Item>
         </Stack>
-        <Stack.Item grow={3} style={bodyStyles}>
+        <Stack.Item grow={3} style={ (canConnect) ? bodyStylesWithConnect : bodyStyles}>
           <Text>{props.resource.properties.description}</Text>
         </Stack.Item>
+        { canConnect &&
```

Comment:
is it normal to put the controls to render on the other side of an `&&`?

Reply:
wouldn't this usually have a code block that returned the content?

Reply:
I agree it looks weird and took me some time to accept :) However, seems to be the/an accepted way to do it. We have it across the codebase already. Here's another example https://github.com/microsoft/AzureTRE/blob/e1a302c145159f80faee32879f9e920dd0a9f981/ui/app/src/components/shared/ResourceCard.tsx#L99

Reply:
it's a shortcut for if is truthy, render the following element. looks weird but saves a bunch of extra lines